bfd/elfnn-riscv.c

```diff
@@ -2877,6 +2885,139 @@ riscv_merge_non_std_and_sv_ext (bfd *ibfd,
   return TRUE;
 }
 
+typedef struct riscv_std_z_ext_mergetab {
+  riscv_subset_t *input[RISCV_STD_Z_EXT_COUNT];
+  riscv_subset_t *output[RISCV_STD_Z_EXT_COUNT];
+} riscv_std_z_ext_mergetab_t;
```

Comment:
I am thinking we might have a common data struct for holding all `s`, `x`, `z` and `h` extensions, so might use a linked list instead of an array might be better, and then we don't need maintain the magic number of array size. 

And then we could share the logic among `s`, `x`, `z` and `h` extensions.

Reply:
I've added some code which now merges all the `{s,sx,z,x}` extensions in canonical order, by generalising the `riscv_merge_non_std_and_sv_ext` function to handle all the classes. I've renamed it to `riscv_merge_non_std_ext`.

There are a couple of regressions now, but only due to error messages changing slightly, or some tests which should no longer pass under the new parsing scheme.